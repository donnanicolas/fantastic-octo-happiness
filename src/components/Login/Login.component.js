// @flow
import React, { Component } from 'react';

import { authUrl } from '../../utils/auth';

import Button from '../Button';
import { FlexContainer, FlexItem } from '../Flex';

class Login extends Component {
  render() {
    return (
      <FlexContainer justifyContent="center">
        <FlexItem>
          <Button href={authUrl} primary>
            Login With Spotify
        </Button>
        </FlexItem>
      </FlexContainer>
    )
  }
}

export default Login;