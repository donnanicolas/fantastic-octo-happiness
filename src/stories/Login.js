// @flow
import React from 'react';
import { storiesOf } from '@storybook/react';

import Login from '../components/Login';

const addStories = () => {
  storiesOf('Login', module)
    .add('default', () => <Login  />)
};

export default addStories;