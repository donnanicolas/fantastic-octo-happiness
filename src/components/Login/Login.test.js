import React from 'react';
import Login from './Login.component';
import renderer from 'react-test-renderer';

it('renders the login page', () => {
  const tree = renderer.create(
    <Login />
  ).toJSON()
  expect(tree).toMatchSnapshot();
});

