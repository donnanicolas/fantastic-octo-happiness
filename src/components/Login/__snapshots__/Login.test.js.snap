// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the login page 1`] = `
<div
  className="gqseAW"
  justifyContent="center"
>
  <div
    className="fSnuyn"
  >
    <a
      className="btn btn-primary"
      href="https://accounts.spotify.com/authorize?client_id=89b92773702c4e36a8014a880e7a9b7d&response_type=token&redirect_uri=http%3A%2F%2Flocalhost%3A3000%2Flogin&scope=user-library-read%20user-read-email%20user-library-modify"
      onClick={[Function]}
    >
      Login With Spotify
    </a>
  </div>
</div>
`;
